### PR TITLE
Add `forward_mut` in `Sequential` derive

### LINKED ISSUE #881: Bug: `Sequential` macro provide `forward_mut` as `forward`
Because of this models like `BatchNorm2D` crashes while training.
https://github.com/coreylowman/dfdx/blob/370334fda03057c0e62e442262cb00e87c06c4c8/dfdx/src/nn/layers/batch_norm2d.rs#L92
https://github.com/coreylowman/dfdx/blob/370334fda03057c0e62e442262cb00e87c06c4c8/dfdx-derives/src/lib.rs#L519-L522

### PR DESCRIPTION
**Pull request by swfsql**
Closes #881

### CODE REVIEW

**Review by chelsea0x3b — APPROVED** [review]
Thanks for this fix!
